### DISCUSSION TITLE
Wordpress problems domain and SSL

### DISCUSSION THREAD

**Discussion by Toninobandolero** [caprover/caprover#1314] [Q&A] [answered]
Hello everyone, and thanks for this great software. I don't have much experience in system administration.
I have installed Caprover on my vps running ubuntu 20.04. I have also installed some one-click apps that work very well for me. But I am having a problem with wordpress. The problem is that by default caprover creates a subdomain within the main domain that you use with the app. In the case of the rest of the apps there is no problem, but with wordpress I need it to be redirected somehow to the maindomain.com To do that, I have connected and activated https in "http settings" a new domain maindomain.com, which also use to caprover. To begin with, I don't know if that's done right. Once connected I change site_url and home_url in wordpress. It seems that everything is fine, but it gives me several problems: if I force the https I get the error "too many redirects" when I enter /wp-admin. If I remove force https, the problem is that it does not enter through https securely. I do not know if I explained well. I hope if someone understands my problem can help me. Thanks a lot :)

**Comment by githubsaturn** [comment]
What you did was correct. Not sure what happened in your case. Here is an example of a fully working version

- I installed wordpress
- I connected a custom domain
- I enabled HTTPS for that domain and all subdomains
- I logged in to CapRover and changed the site URL
- I enforced HTTPS in CapRover settings
![image](https://user-images.githubusercontent.com/10582082/150545266-1bfa4569-733b-4cd1-9a70-d6176d100609.png)


You can see the website at
https://custom-wp.demo.caprover.com

and you can see the admin login https://custom-wp.demo.caprover.com/wp-admin


You can log in to see the config:
- URL: https://captain.server.demo.caprover.com
- Password `captain42`

**Reply by Toninobandolero** [reply]
Hello. Thank you very much for your answer and your help. I have followed the steps you have described, but the result is the same again:

This page does not workThe page codigoabierto.xyz has redirected you too many times.
Delete cookies.

ERR_TOO_MANY_REDIRECTS

You can see it here:

codigoabierto.xyz

**When i take out "Force HTTPS by redirecting all HTTP traffic to HTTPS" it just work....** Could be a DNS problem?

what do you mean with "I enabled HTTPS for that domain **and all subdomains**" how do you do that?

why are you using http and https in your wp config?

**Reply by leninriv** [reply]
did you where able to fix this?
i having this issue when i connect my custom domain to my caprover wordpress aplicaion

**Reply by malipetek** [reply]
Do you use cloudflare by any chance?
https://vercel.com/guides/resolve-err-too-many-redirects-when-using-cloudflare-proxy-with-vercel

**Comment by githubsaturn** [comment, marked as answer]
> why are you using http and https in your wp config?

Just a typo. I fixed it, and it still works.

> what do you mean with "I enabled HTTPS for that domain and all subdomains" how do you do that?

![image](https://user-images.githubusercontent.com/10582082/150646552-4cfa2b6b-8e45-4cf7-b7ad-48ba023c43d6.png)

> When i take out "Force HTTPS by redirecting all HTTP traffic to HTTPS" it just work.... Could be a DNS problem?

Yes. Specially if you're using a CDN like cloudflare etc.

**Reply by Toninobandolero** [reply]
Yes, i am using cloudflare. Do you recommend me not to use it?
How could i fix the problem from cloudflare?

Thanks :)

**Reply by githubsaturn** [reply]
You'll have to figure out what Cloudflare does that causes this. I'm unsure what they do. But you're not using a plain DNS, that's the problem. You're using a proxy. Using a proxy may work, but it causes weirdness like this.

**Reply by Toninobandolero** [reply]
Thank you!
I just use the cloudflare's option to Redirect all requests with the combination "http" to "https". It applies to all HTTP requests from the zone for my domain, and it seems to work. It is somewhat more convoluted, but in the end it has worked. Thank you for your quick response and for your amazing software.
